src/Raven.Server/Documents/Queries/Graph/SingleEdgeMatcher.cs

```diff
@@ -44,27 +45,59 @@ public void SingleMatch(Match left, string alias)
                 if (BlittableJsonTraverser.Default.TryRead(leftDoc, Edge.Path.FieldValue, out var value, out _) == false)
                     return;
 
-                if (Edge.Project == null)
+                //allow array of primitives
+                if (Edge.Project == null && !(value is BlittableJsonReaderArray))
                 {
                     ThrowMissingEdgeProjection();
                 }
 
+                var projectFieldValue = Edge.Project?.FieldValue;
                 switch (value)
                 {
                     case BlittableJsonReaderArray array:
                         foreach (var item in array)
                         {
-                            if (item is BlittableJsonReaderObject json &&
-                                Edge.Where?.IsMatchedBy(json, QueryParameters) != false)
+                            switch (item)
                             {
-                                AddEdgeAfterFiltering(left, json, Edge.Project?.FieldValue);
+                                case BlittableJsonReaderObject json when Edge.Where?.IsMatchedBy(json, QueryParameters) != false:
+                                    if (projectFieldValue == null)
+                                        ThrowMissingEdgeProjection();
+                                    AddEdgeAfterFiltering(left, json, projectFieldValue);
+                                    break;
+                                case LazyStringValue lazyString:
+                                    var jsonStringForWhere = new DynamicJsonValue
```

Comment:
Don't we need to check that `projectFieldValue` is null here?

Reply:
Probably, and also we need to test for arrays of primitives that are not strings.
I did this PR because I am not 100% sure that the code so far is good enough - and I mean the code around IsMatchedBy() that I added